### PR TITLE
Feature branch: ORA staff grading

### PR DESCRIPTION
**Pull request by andy-armstrong**
Sandbox: https://cahrens.sandbox.edx.org/courses/course-v1:ora+ora+ora/courseware/8d9584d242b44343bc270ea5ef04ab03/

https://openedx.atlassian.net/browse/TNL-3461

### CODE REVIEW

File: openassessment/workflow/models.py [hunk 1]
```diff
@@ -97,6 +96,20 @@ class Meta:
         ordering = ["-created"]
         # TODO: In migration, need a non-unique index on (course_id, item_id, status)
 
+    def __init__(self, *args, **kwargs):
```
Comment (efischer19) [inline openassessment/workflow/models.py]: @cahrens context for your error we're discussing

**Comment by cahrens** [comment]
@cptvitamin This feature branch is ready for accessibility review. The branch is on my sandbox (https://cahrens.sandbox.edx.org), and devops has set the sandbox so it will not expire until mid-January.

A few things to note:

1) When logged in as staff, you can do the "staff override" workflow on any ORA problem by searching for the learner's name in "Manage Individual Users", and then expanding "Submit Assessment Grade Override". Currently staff, honor, verified, and audit are all enrolled in the course, but they won't have submitted responses for every problem.1

2) The course has 2 problems set up that REQUIRE staff grading (the "full grade" workflow). These are https://cahrens.sandbox.edx.org/courses/course-v1:ora+ora+ora/courseware/61944efb38a349edb140c762c7419b50/415c3ee1b7d04b58a1887a6fe82b31d6/ and https://cahrens.sandbox.edx.org/courses/course-v1:ora+ora+ora/courseware/8d9584d242b44343bc270ea5ef04ab03/0b0dcc728abe45138c650732af178afb/ On these problems, you will see the additional "Grade Available Responses" staff tool.

3) You can clear a student's submission state so that you can easily resubmit submissions and try parts of the workflow over. You do this by copying the xblock Location in the "View Assignment Statistics" staff tool, then going over to the Instructor tab, Student Admin subsection. Put the student's name in Student-Specific Grade Adjustment, paste in the problem location, and press "Delete Student State for Problem".

image

@dianakhuang is around next week and can answer questions. Have fun reviewing!

File: openassessment/xblock/static/js/src/lms/oa_staff_area.js [hunk 2]
```diff
@@ -148,117 +279,222 @@
...
+         *     classes in different form).
+         * @param {boolean} continueGrading If true, another learner will be marked as "In Progress",
+         *     and a new grading form will be rendered with the learner's answer.
+         */
+        submitStaffGrade: function(submissionID, rubric, scope, continueGrading) {
+            var view = this;
+            var successCallback = function() {
+                view.baseView.unsavedWarningEnabled(false, view.FULL_GRADE_UNSAVED_WARNING_KEY);
+                view.staffGradeFormLoaded = false;
+                view.loadStaffGradeForm(!continueGrading);
```
Comment (cahrens) [inline openassessment/xblock/static/js/src/lms/oa_staff_area.js]: @cptvitamin, I realized after setting up the sandbox that we need to scroll up in the "continueGrading" case or else the staff member can't tell that a new form has been loaded for grading. I have added that scroll operation in #827. I still wonder what we need to do with focus though.

File: openassessment/templates/openassessmentblock/oa_base.html [hunk 3]
```diff
@@ -30,7 +30,7 @@ <h2 class="step__title">
                             <span class="step__status">
                                 <span class="step__status__label">{% trans "This step's status:" %}</span>
                                 <span class="step__status__value">
-                                  <i class="icon fa fa-spinner fa-spin"></i>
+                                  <i class="icon fa fa-spinner fa-spin" aria-hidden="true"></i>
```
Comment (cptvitamin) [inline openassessment/templates/openassessmentblock/oa_base.html]: thanks for adding this!

File: openassessment/templates/openassessmentblock/edit/oa_edit_staff_assessment.html [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{% load i18n %}
+{% spaceless %}
+<li class="openassessment_assessment_module_settings_editor" id="oa_staff_assessment_editor">
+    <div class="drag-handle action"></div>
+    <div class="openassessment_inclusion_wrapper">
+        <input type="checkbox" id="include_staff_assessment"
```
Comment (cptvitamin) [inline openassessment/templates/openassessmentblock/edit/oa_edit_staff_assessment.html]: this control is inaccessible. beyond scope of this PR.  new ticket here: https://openedx.atlassian.net/browse/ECOM-3389

File: openassessment/templates/openassessmentblock/grade/oa_grade_complete.html [hunk 5]
```diff
@@ -1,337 +1,197 @@
...
+                        {% include "openassessmentblock/oa_uploaded_file.html" with file_upload_type=file_upload_type file_url=file_url header="Your Upload" class_prefix="submission__answer" %}
+                    </article>
+
+                    <article class="submission__peer-evaluations step__content__section">
+                        <h3 class="submission__peer-evaluations__title">{% trans "Assessments of Your Response" %}</h3>
+
+                        <ol class="list submission__peer-evaluations__questions">
+                            {% for criterion in grade_details.criteria %}
+                                {% with criterion_num=forloop.counter %}
+                                    <li class="question question--{{ criterion_num }} ui-toggle-visibility">
```
Comment (cptvitamin) [inline openassessment/templates/openassessmentblock/grade/oa_grade_complete.html]: beyond scope of this PR, but anything with ui-toggle-visibility is not keyboard accessible and contents are not actually hidden in the DOM (i think they are given a zero height): https://openedx.atlassian.net/projects/AC/issues/AC-316

**Pull request merged**